scipy/stats/_distribution_infrastructure.py

```diff
@@ -2501,11 +2499,11 @@ def _cdf_complement(self, x, **params):
         return 1 - self._ccdf_dispatch(x, **params)
 
     def _cdf_complement_safe(self, x, **params):
-        out = self._cdf_complement(x, **params)
+        ccdf = self._ccdf_dispatch(x, **params)
+        out = 1 - ccdf
         eps = np.finfo(self._dtype).eps
         tol = self.tol if not _isnull(self.tol) else np.sqrt(eps)
-        with np.errstate(divide='ignore'):
-             mask = tol < eps/out
+        mask = tol * out < np.spacing(ccdf)
```

Comment:
`eps` was a conservative estimate of `spacing(ccdf)`. I think it's better to use `spacing`, though.